contracts/staking/GoodFundManager.sol

```diff
@@ -189,103 +221,191 @@ contract GoodFundManager is DAOContract {
 		blockInterval = _blockInterval;
 	}
 
-	/**
-	 * @dev Checks if enough time has passed away since the
-	 * last funds transfer time
-	 * @return (bool) True if enough time has passed
-	 */
-	function canRun() public view returns (bool) {
-		return block.number.div(blockInterval) > lastTransferred;
-	}
-
 	/**
 	 * @dev Collects UBI interest in iToken from a given staking contract and transfers
 	 * that interest to the reserve contract. Then transfers the given gd which
 	 * received from the reserve contract back to the staking contract and to the
 	 * bridge, which locks the funds and then the GD tokens are been minted to the
 	 * given address on the sidechain
-	 * @param _staking Contract that implements `collectUBIInterest` and transfer iTokeb to
-	 * a given address. The given address should be the same whitelisted `reserve`
-	 * address in the current contract, in case that the given staking contract transfers
-	 * the funds to another contract, zero GD tokens will be minted by the reserve contract.
-	 * Emits `FundsTransferred` event in case which interest has been passed to the `reserve`
 	 */
-	function transferInterest(StakingContract _staking)
-		public
-		reserveHasInitialized
-	//requireDAOContract(address(_staking))
-	{
-		// require(
-		//     canRun(),
-		//     "Need to wait for the next interval"
-		// );
-        address reserveAddress = nameService.addresses(nameService.RESERVE());
+	function collectInterest(address[] memory _stakingContracts) public {
+		uint256 initialGas = gasleft();
+		_reserveHasInitialized();
 		lastTransferred = block.number.div(blockInterval);
-		ERC20 iToken = ERC20(_staking.iToken());
+		ERC20 iToken = ERC20(nameService.getAddress("CDAI"));
 		// iToken balance of the reserve contract
-		uint256 currentBalance = iToken.balanceOf(reserveAddress);
-		// collects the interest from the staking contract and transfer it directly to the reserve contract
-		//`collectUBIInterest` returns (iTokengains, tokengains, precission loss, donation ratio)
-	    _staking.collectUBIInterest(reserveAddress);
+		uint256 currentBalance =
+			iToken.balanceOf(nameService.addresses(nameService.RESERVE()));
```

Comment:
read reserve address to temporary variable, reading from nameservice many times is gas expensive